sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/uninstall/UninstallScheduler.java

```diff
@@ -34,20 +34,19 @@
     private OfferAccepter offerAccepter;
 
     /**
-     * Creates a new UninstallScheduler based on the provided API port and initialization timeout,
-     * and a {@link StateStore}. The UninstallScheduler builds an uninstall {@link Plan} with two {@link Phase}s:
-     * a resource phase where all reserved resources get released back to Mesos, and a deregister phase where
-     * the framework deregisters itself and cleans up its state in Zookeeper.
+     * Creates a new {@link UninstallScheduler} based on the provided API port and initialization timeout, and a
+     * {@link StateStore}. The {@link UninstallScheduler} builds an uninstall {@link Plan} which will clean up the
+     * service's reservations, TLS artifacts, zookeeper data, and any other artifacts from running the service.
      */
     public UninstallScheduler(
             ServiceSpec serviceSpec,
             StateStore stateStore,
             ConfigStore<ServiceSpec> configStore,
             SchedulerFlags schedulerFlags,
-            Optional<SecretsClient> secretsClient) {
+            Optional<SecretsClient> customSecretsClientForTests) {
```

Comment:
Could you just refactor to a second protected constructor that takes the secret client?

Reply:
Good point, done